test/Test.java

```diff
@@ -83,4 +85,47 @@ public String innerString() {
       }
   }
 
+  public int getInt() {
+    return integerField;
+  }
+
+  public void setInt(int v) {
+    integerField = v;
+  }
+
+  public String getString() {
+    return stringField;
+  }
+
+  public void setString(String s) {
+    stringField = s;
+  }
+
+  public Object getObject() {
+    return objectField;
+  }
+
+  public void setObject(Object val) {
+    objectField = val;
+  }
+
+  public String toString() {
+      return "Test(" + integerField + ", " + stringField + ")";
+  }
+
+  public String testArrayArgs(int i) {
+    return "int";
+  }
+
+  public String testArrayArgs(int[] i) {
+    return "int[]";
+  }
+
+  public String testArrayArgs(int[][] i) {
+    return "int[][]";
+  }
+
+  public String testArrayArgs(Object[] i) {
+    return "java.lang.Object[]";
+  }
```

Comment:
The changes to Test.java and the binaries should be reverted for now and moved into a separate sub-package.